compiler/rustc_driver/src/lib.rs

```diff
@@ -162,6 +163,8 @@ pub fn run_compiler(
     let sopts = config::build_session_options(&matches);
     let cfg = interface::parse_cfgspecs(matches.opt_strs("cfg"));
 
+    let mut make_codegen_backend = Some(make_codegen_backend);
```

Comment:
Maybe add a comment saying that we wrap it in another `Option` so that `dummy_config` can take ownership when needed, but we also still can create the full `config` below?
It took me a bit to understand what is going on here.

Reply:
Done